### PR TITLE
Add different samplerate samples

### LINKED ISSUE #1022: Transcode input audio to 16 kHz, 16-bit depth in the Python client
Both the C++ and the JS clients leverage SoX to do this, but the Python client doesn't. We should make them consistent.

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #1022

### CODE REVIEW

File: native_client/client.cc [hunk 1]
```diff
@@ -89,17 +89,17 @@ main(int argc, char **argv)
...
   sox_signalinfo_t target_signal = {
-      SOX_UNSPEC, // Rate
+      16000, // Rate
       1, // Channels
       16, // Precision
       SOX_UNSPEC, // Length
       NULL // Effects headroom multiplier
   };
 
+  sox_signalinfo_t interm_signal;
```
Comment (lissyx) [inline native_client/client.cc]: Coming from sox' examples

File: native_client/python/client.py [hunk 2]
```diff
@@ -37,20 +37,19 @@
 N_CONTEXT = 9
 
 def convert_samplerate(audio_path):
-    sox_cmd = 'sox --norm {} -b 16 -t wav - channels 1 rate 16000'.format(audio_path)
+    sox_cmd = 'sox {} --type raw --bits 16 --channels 1 --rate 16000 - '.format(audio_path)
```
Comment (lissyx) [inline native_client/python/client.py]: Force raw in-par with NodeJS processing, and removes the need for `split('data')`

File: taskcluster/test-linux-opt-base.tyml [hunk 3]
```diff
@@ -53,7 +53,7 @@ then:
           extraSystemSetup: { $eval: strip(str(build.system_setup)) }
           installGitlfs: { $eval: strip(str(build.git_lfs.linux)) }
         in: >
-          apt-get -qq update && apt-get -qq -y install git pixz libsox2 wget && ${extraSystemSetup} &&
+          apt-get -qq update && apt-get -qq -y install git pixz sox wget && ${extraSystemSetup} &&
```
Comment (lissyx) [inline taskcluster/test-linux-opt-base.tyml]: sox binary will bring libsox

**Comment by lissyx** [comment]
@reuben We might need to verify a bit closely how intermittent failures can be on the "prod" set. Also we should discuss what we do with upsampling. The current 8kHz sample in the repo is down-sampled from the 16kHz one. So when we up-sample this in the binaries, then we loose too much data.

**Comment by reuben** [comment]
We lose data when we downsample, not when we upsample. Maybe we should error out or print a warning when the input is <16kHz, instead of silently converting and giving a bad output.

Having intermittent failures in these tests would be pretty bad, maybe we can test only that it runs and outputs *something*, instead of checking for exactly that wash water string?

**Comment by lissyx** [comment]
@reuben Yes, my wording was inconvenient, I meant that when we upsample in the binaries, we cannot recover what we lost :). I'm open to any suggestion, but maybe erroring would be too much?

For the intermittents tests, we could do that.

**Comment by reuben** [comment]
I'd be happy with a "Warning: Upsampling input from N kHz to 16 kHz, transcription accuracy may be affected." or something like that.

**Comment by lissyx** [comment]
@reuben Tests should be better now. For the warning, though, we might need more work: the NodeJS code does takes the input file directly without looking at it, so we cannot check with the current code if the bitrate is lower than 16kHz. I'll try and change that.

File: native_client/javascript/client.js [hunk 4]
```diff
@@ -44,6 +45,13 @@ function totalTime(hrtimeValue) {
...
 
+const buffer = Fs.readFileSync(args['audio']);
+const result = Wav.decode(buffer);
+
+if (result.sampleRate < 16000) {
+  console.error('Warning: original sample rate (' + result.sampleRate + ') is lower than 16kHz. Up-sampling might produce erratic speech recognition.');
+}
+
 var audioStream = new MemoryStream();
 Fs.createReadStream(args['audio']).
```
Comment (reuben) [inline native_client/javascript/client.js]: This is now reading the file twice. You can do this to reuse the existing buffer: https://gist.github.com/reuben/a8268789bdcc4dcfb5cecc61829848c9

Comment (lissyx) [reply native_client/javascript/client.js]: Awesome, I wanted to do that but I did fail.

**Comment by lissyx** [comment]
Failures were because macbooks are too busy. Retriggers are green.

**Pull request merged**